x-pack/plugins/actions/server/builtin_action_types/lib/get_proxy_agents.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Agent } from 'http';
+import HttpProxyAgent from 'http-proxy-agent';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { Logger } from '../../../../../../src/core/server';
+import { ProxySettings } from '../../types';
+
+interface GetProxyAgentsResponse {
+  httpAgent: Agent | undefined;
+  httpsAgent: Agent | undefined;
+}
+
+export function getProxyAgents(
+  proxySettings: ProxySettings | undefined | null,
+  logger: Logger
+): GetProxyAgentsResponse {
+  const undefinedResponse = {
+    httpAgent: undefined,
+    httpsAgent: undefined,
+  };
+
+  if (!proxySettings) {
+    return undefinedResponse;
+  }
+
+  logger.debug(`Creating proxy agents for proxy: ${proxySettings.proxyUrl}`);
+  let proxyUrl: URL;
+  try {
+    proxyUrl = new URL(proxySettings.proxyUrl);
+  } catch (err) {
+    logger.warn(`invalid proxy URL "${proxySettings.proxyUrl}" ignored`);
```

Comment:
**nit:** Should this message include `err.message` or its pretty clear this is the information each time an error is thrown?

Reply:
From the REPL, the only message I see coming out of bad URL contructor calls is

```console
> new URL('xyz://90:80\\')
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL: xyz://90:80\
```

Given that state of usefulness, seems like the current code is good enough :-)